test/sequential/market.nonTransferable.multiswap.test.ts

```diff
@@ -0,0 +1,294 @@
+import { jest } from "@jest/globals";
+import { BN } from "ethereumjs-util";
+import { getApi, initApi } from "../../utils/api";
+import { Assets } from "../../utils/Assets";
+import { FoundationMembers } from "../../utils/FoundationMembers";
+import { getSudoUser, setupApi, setupUsers } from "../../utils/setup";
+import { User } from "../../utils/User";
+import { BN_BILLION } from "@polkadot/util";
+import { Sudo } from "../../utils/sudo";
+import { getEventResultFromMangataTx } from "../../utils/txHandler";
+import {
+  ExtrinsicResult,
+  waitSudoOperationFail,
+  waitSudoOperationSuccess,
+} from "../../utils/eventListeners";
+import { Market } from "../../utils/market";
+import { GASP_ASSET_ID } from "../../utils/Constants";
+import {
+  BN_HUNDRED_BILLIONS,
+  BN_MILLION,
+  BN_TEN_THOUSAND,
+  BN_ZERO,
+  MangataGenericEvent,
+} from "gasp-sdk";
+import { getLiquidityAssetId } from "../../utils/tx";
+import { ApiPromise } from "@polkadot/api";
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(2500000);
+process.env.NODE_ENV = "test";
+
+let api: ApiPromise;
+let sudo: User;
+let testUser: string;
+let token1: BN;
+let token2: BN;
+let token3: BN;
+let token4: BN;
+let token5: BN;
+let token6: BN;
+let liqIds: BN[];
+
+async function getStablePoolId(events: MangataGenericEvent[]) {
+  const poolIds: BN[] = [];
+  let poolId: BN;
+  let i = 0;
+  const poolCreatedEvents = JSON.parse(
+    JSON.stringify(events.filter((events) => events.method === "PoolCreated")),
+  );
+  const length = poolCreatedEvents.length;
+  while (i < length - 1) {
+    poolId = new BN(poolCreatedEvents[i].event.data[1]);
+    poolIds.push(poolId);
+    i++;
+  }
+  return poolIds;
+}
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+
+  await setupApi();
+  await setupUsers();
+  sudo = getSudoUser();
+  testUser = "0x798d4ba9baf0064ec19eb4f0a1a45785ae9d6dfc";
+  api = getApi();
+});
+
+beforeEach(async () => {
+  let foundationMembers: any;
+
+  foundationMembers = await FoundationMembers.getFoundationMembers();
+
+  [token1, token2, token3, token4, token5, token6] =
+    await Assets.setupUserWithCurrencies(
+      sudo,
+      [BN_BILLION, BN_BILLION, BN_BILLION, BN_BILLION, BN_BILLION, BN_BILLION],
+      sudo,
+    );
+
+  const oldFounder2 = foundationMembers[2];
+  await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      oldFounder2,
+      api.tx.foundationMembers.changeKey(testUser),
```

Comment:
reuse the class created on previous PR :)

Reply:
Done, the function was changed